src/WebJobs.Script.WebHost/Controllers/HostController.cs

```diff
@@ -208,6 +212,8 @@ public async Task<IActionResult> SetState([FromBody] string state)
 
                 // we're currently online and the request is to take the host offline
                 await FileMonitoringService.SetAppOfflineState(_applicationHostOptions.Value.ScriptPath, true);
+                // TODO: pgopa shutdown rpcserver
+                await _languageWorkerChannelManager.ShutdownChannelsAsync();
```

Comment:
TODO: add tests.

@mathewc  - is this the right place to shutdown rpc resources?

Reply:
Actually you can remove this - you don't have to do anything when we go offline, because it results in the process shutting down (code [here](https://github.com/Azure/azure-functions-host/blob/2ef5d7223e5d2e2c44ca54e3becfa840af9303f5/src/WebJobs.Script.WebHost/FileMonitoringService.cs#L154)). The WebApp will then restart in Offline mode - in Offline mode you shouldn't start the language workers.

Reply:
great. thanks!